pyop2/compilation.py

```diff
@@ -200,17 +201,22 @@ def workaround_cflags(self):
         return []
 
     @collective
-    def get_so(self, src, extension):
+    def get_so(self, jitmodule, extension):
```

Comment:
I want to check the caching inside `get_so` (and defer code generation).

Reply:
OK, so the idea is that you don't want to have to generate the source code to determine the hash.

I think I would prefer somehow the JITModule managing the hashing and cache lookup. But perhaps we should open an issue for subsequent improvements rather than trying to fix everything now.

Reply:
Done.